viewer_controls.md

```diff
@@ -0,0 +1,296 @@
+ - toggles channel on/off
+- Action
+ - trigger a "render" image event
+ - trigger a modification of the enabled flag of the Save button
+ - trigger a modification of the enabled flag of the Undo button
+
+**Slider**
+- At Event - move left-right knobs
+ - sets the input intensity interval
+ - updates values in text fields on both side of the slider
+- Action - when slider released
+ - trigger a "render" image event
+ - trigger a modification of the enabled flag of the Save button
+ - trigger a modification of the enabled flag of the Undo button
+
+ **Text field**
+- Action - when Return hit or focus lost
+ - update slider's value
+ - trigger a "render" image event
+ - trigger a modification of the enabled flag of the Save button
+ - trigger a modification of the enabled flag of the Undo button
+
+**Color picker**
+- At Event
+ - open color-picker dialog
+
+#### Color picker dialog
+
+**Set-up: LUT loading**
+- lut are loaded using the method ``listLuts_json`` see [views.py](<https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroWeb/omeroweb/webgateway/views.py>)
+- currently the icon associated to a lut is determined using a pre-build image. (To be modified)
+
+**Toolbar button - color wheel**
+- Action
+ - changes selector pane to color wheel selector
+
+ **Toolbar button - rgb**
+- Action
+ - changes selector pane to rgb selector
+
+ **Toolbar button - default color picker button**
+- Action
+ - changes selector pane to default color picker selector
+
+**Color/lut selection**
+- Default value
+ - set using the retrieved settings
+- At Event
+ - selects color/lut
+- Action
+ - enable Preview, Accept, Revert buttons
+
+**Alpha slider**
+- Default value
+ - set using the retrieved settings
+- At Event - move knob
+  - sets alpha value
+  - updates value in text fields on right of the slider
+- Action
+ - enable Preview, Accept, Revert buttons
+
+**Preview button**
+- Disabled
+ - default
+- Enabled
+ - when the settings have been changed
+- Action
+ - trigger a "render" image event
+ - trigger a modification of the enabled flag of the Save button
+ - trigger a modification of the enabled flag of the Undo button
+
+**Accept button**
+- Disabled
+ - default
+- Enabled
+ - when the settings have been changed
+- Action
+ - trigger a "render" image event
+ - trigger a modification of the enabled flag of the Save button
+ - trigger a modification of the enabled flag of the Undo button
+ - background of the button and the slider updated
+ - close color picker dialog
+
+**Cancel button**
+- Enabled
+ - all the time
+- Action
+ - revert to rendering settings at time of opening of color picker 
+ - update color picker display
+ - trigger rendering settings update - ? only if Preview had been used
+
+#### Controls for all channels
+
+**Greyscale checkbox**
+- Unchecked
+ - default
+- Event - check
+ - toggles all on channels off
+- set rendering to grey scale
+- Event - uncheck
+ - toggles all channels that were on back on
```

Comment:
In web, we don't keep track of what channels were on before greyscale, so we still only have a single channel on after reverting greyscale -> color.